### PR TITLE
Replace CAS loop by updateAndGet to improve readability

### PR DESCRIPTION
**Pull request by dnhatn**
Replace a CAS tight-loop by `AtomicLong#updateAndGet` to improve readability.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM, yet please wait for a review from @s1monw too.

**Review by s1monw — APPROVED** [review]
LGTM too

**Comment by dnhatn** [comment]
Thanks @jasontedor and @s1monw.

**Pull request merged**
